src/metadata.ts

```diff
@@ -34,5 +43,5 @@ export function normalizeMetadata(input?: File | Response | BufferLike | StreamL
   if (input instanceof Blob) return { encodedName, uncompressedSize: BigInt(input.size) }
   if (input instanceof ArrayBuffer || ArrayBuffer.isView(input)) return { encodedName, uncompressedSize: BigInt(input.byteLength) }
   // @ts-ignore
-  return { encodedName, uncompressedSize: size > -1 ? BigInt(size) : undefined }
+  return { encodedName, uncompressedSize: size > -1 ? BigInt(size) : BigInt(0) }
```

Comment:
I'm not comfortable with this. Previously, if you wanted to predict the length of an archive, returning `undefined` here ensured that an error would occur if we couldn't determine the length of a file. Instead with your change, the file's size will be silently set to zero, resulting in an incorrect size prediction.

I would prefer a more explicit way to add empty folders to the archive, where we can see the user's intention and not just guess. For example, the object could include `folder: true` or `type: 'folder'`. This would also allow properly setting the relevant UNIX file type and mod bitmasks, and decide if we really want to add a trailing slash. After all, it is also entirely possible that the user wanted to create an empty file (for which the size should be explicitly zero).

Reply:
Alright. So if you want an empty file you pass some empty input like `""`, or `size: 0` in the metadata, and if you want an empty folder you leave both the input and size `undefined`. Works for me.

That logic needs to be used everywhere (for example, when you ensure the trailing slash). Don't assume that just because the input was undefined, it's meant to be a folder. We're processing metadata there, so the user could pass an object with a *size* and no *input*.